### DISCUSSION TITLE
Is there a way to extend markdown posts with VueJS outside of the .md file?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vuejs/vitepress#1137] [Q&A] [answered]
Is there a way to keep my markdown files purely markdown but customize the theme/layout to add Vue components behind the scenes?

I'd like to keep my markdown files purely markdown for ease of use between apps that I use for writing, blogging, etc.

For example, every post I write has a `Newsletter` and `Comments` component added to the bottom of the .md file. I'd prefer to maintain this using and/or extending a layout file rather than having to add it to each markdown file.

**Comment by brc-dd** [comment, marked as answer]
You can use slots for that: https://vitepress.vuejs.org/guide/theme-introduction.html#layout-slots

**Reply by CodyBontecou** [reply]
Perfect!
